reproject/interpolation/core_full.py

```diff
@@ -126,10 +127,22 @@ def _reproject_full(array, wcs_in, wcs_out, shape_out, order=1):
 
     coordinates = pixel_in.transpose()[::-1]
 
-    array_new = map_coordinates(array,
+    if array_out is not None:
+        assert array_out.shape == shape_out
```

Comment:
`asserts` also don't play nice in tests when `pytest` is used (don't know if it's used for `reproject`) and should be avoided.